### ISSUE TITLE
Proposal 4 little new function

### ISSUE THREAD

**Issue by Carpspb** [lairdubois/lairdubois-opencutlist-sketchup-extension#188]
Hey there,
Dear Boris,
I have a proposal 4 the openculist export function:
Explanation: In Sketchup (nearly) erverybody knows you have the possibility to define attributes on every component. One can do this by adding inside a component an attribute "componentattributes" as a dynamic component. You can put there userdefined text or even make calculations etc....
With the sketchup "generate report" you can write everything in an csv-table....just as one can do in opencutlist with width,height,length,material,name (the thing with the name is really cool!!).
Now my proposal: extend the funktionality of opencutlist with the possibility to export these previosly in sketchup dynamic component defined attribute fields for the opencutlist csv report . I imagine it like this: Go to the settings in open cutlist, type in a user-defined attribute field (naturellement, the same like the attribute field in sketchup), perhaps user defined descriptions or usertext or isthiswoodpiecebeautiful, likeEuropeYN, remarkABC, or instructionABC, assemblingrequirements ....
this would give opencutlist the possibility not only to export woodlist / partlist but also an additional field for example for descriptions or instructions etc. (like above mentioned) or whatever. Perhaps even more fields
Like this first line in the OCL-report:
nr.|path|instanc|definition|lengt|width|thicknes|material|etiquette... <---provided yet
and additionally the attribute fields:
remarkABC|instructionABC|
The usage could also be for engeneering bureaus for item lists belonging to the item plans or carpenter or joiner or cabinet maker companies for production notes et ceterapepe

Perhaps there could be the possibility to read out userdefined dictionaries (not the dynamic component ones): In sketchup you have the possibility to define dictionaries and attributes with several other extensions like attributeinspektor from aerilius. I know there several other extensions where you can create attributes or even manage them. perhabs it could also be possible to create an access to these attributes with OCL
That would be great.
Hoping you understand my intent. Thanks a lot and a good day to you and thank you for doing your "OCL-mission"
carpspb

**Comment by bbeaulant** [comment]
Hi Carpspb,

Thank you to like OCL 👍 
If I understand, your proposal is to export dynamic component custom attributes in OCL export ?

**Comment by Carpspb** [comment]
Yes, brought perfekt on the point for the dynamic_components dictionary. That alone would already be great!
...perhaps it could also be possible for a custom manually self-created dictionary, where it is possible to select in the OCL-settings, then select an attribute-field.
That would be really great if it could come in one of the future versions.
Thank you for your work!

**Comment by bbeaulant** [comment]
Ok, but custom attributes are defined on component definitions. Then multiple part corresponds to multiple definitions that can have multiple non compatible custom attributes.
It's quite particular to merge all of theses attributes into a single table, no ?

**Comment by Carpspb** [comment]
Hey Boris,
thank you for your reply.
Yes the easiest way would be to assign manually attributes over dynamic_attributes through component attributes. Each component in the model must have the same dictionary and "label" like in my FIRST example "dynamic_component_customatrC. Then you can read this string and transfer it in your OCL-Report for each component of the model
![PossibilityONEover_dynamic_attributes](https://user-images.githubusercontent.com/46330843/96268073-e54f7e00-0fc8-11eb-8555-9729a0a2971f.jpg)

Another possiblility would be to assign a dictionary and an attribute to a component over aerelius Attributinspektor (see my SECOND example
![PossibilityTWOover_dynamic_attributes](https://user-images.githubusercontent.com/46330843/96268832-d3baa600-0fc9-11eb-9551-bd75c5c7f74e.jpg)

When one looks that each component has the same dictionary name and the same label (in the second example custom_dictionary_attrA "j'aime OCL") then ----> use it in OCL-Report


Finally the THIRD example:
with the Aerelius Attriubteinspector I could even assignt an attribute to the OCL-dictionary ladb_opencutlist: OCL_CommentAttribute "This is an OCL-Attribute / My comment" ----> use it in OCL-Report

![PossibilityTHREEover_ladb_obencutlist](https://user-images.githubusercontent.com/46330843/96268883-e1702b80-0fc9-11eb-812f-16d4bac7ef76.jpg)

One has the possibility to assign dictionaries / attributes / strings to components over RubyConsole but thats not practikable for everyone. I think the most comfortable way to assign attributes would be over a HTMLinputbox inside skechup or inside the OCL-extension. Perhaps like this: Assigtn to all component the dictionary custom_dictionary and then assign to all components this attribute. And afterwards assignt no component A and B String1 and to component C String2... and/or to a specific selection only
Then in OCL select an in the skechup model existing dictionary and afterwards a attribute and then add this as nex column in OCL-report. Perhaps a second column two...
...my talent for explaining things is not the best i think....
Thank you :-))

**Comment by mazzo80** [comment]
Ciao Carpspb 
Molto interessante anche io ho la necessità di importare altri attributi del componente dinamico....
Quindi è possibile farlo con qualche trucco nella versione che c'è ora?
Scusa ma sono alle prime armi

**Comment by bbeaulant** [comment]
Since version 7.0, DC native attributes and custom attributes could be displayed in export, cutting diagram, labels and drawing.

![Image](https://github.com/user-attachments/assets/8a3a9c8e-61ab-43c8-8950-ecb4538e3e8a)
